pkg/network/server.go

```diff
@@ -219,18 +220,15 @@ func newServerFromConstructors(config ServerConfig, chain Ledger, stSync StateSy
 			}, s.notaryFeer)
 		})
 	}
-	s.bQueue = bqueue.New(chain, log, func(b *block.Block) {
-		s.tryStartServices()
-	}, bqueue.DefaultCacheSize, updateBlockQueueLenMetric, bqueue.NonBlocking)
+	s.bQueue = bqueue.New[*block.Block](chainBlockQueueAdapter{chain}, log, func(b bqueue.Queueable) { s.tryStartServices() }, bqueue.DefaultCacheSize, updateBlockQueueLenMetric, bqueue.NonBlocking)
```

Comment:
> func(b bqueue.Queueable)

It's not. You need to practice with generics and learn the specification. It's `func(b *block.Block)`